src/pages/signin/LoginForm.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import {Text, TextInput, View} from 'react-native';
+import styles from '../../styles/styles';
+import SubmitButton from './SubmitButton';
+import openURLInNewTab from '../../libs/openURLInNewTab';
+import {hasAccount} from '../../libs/actions/Session';
+
+class LoginForm extends React.Component {
+    constructor() {
+        super();
+
+        this.validateAndSubmitForm = this.validateAndSubmitForm.bind(this);
+
+        this.state = {
+            formError: false,
+            login: '',
+            isLoading: false,
+        };
+    }
+
+    /**
+     * Check that all the form fields are valid, then trigger the submit callback
+     */
+    validateAndSubmitForm() {
+        if (!this.state.login.trim()) {
+            this.setState({formError: 'Please fill out all fields'});
+            return;
+        }
+
+        this.setState({
+            formError: null,
+            isLoading: true,
```

Comment:
NAB, I'd expect this to maybe appear as a `prop` set by the API command as a best practice. If something fails and we do not redirect then this will never be set back to `false`.

Reply:
We used to have `session.loading` stored in Onyx, but I thought it was better for each form to control its own state. I don't feel too strongly about it though and could really go either way.

Reply:
It would be great if we could aim for consistency here. Considering that loading states usually are tied to some kind of async action it could make sense to hook these into props passed via `Onyx`. 

We aren't going to be directly handling the result of any action's API request because `Onyx`. And I'm not really sure what the best solution would be but having one way loading states that are not longer loading because you unmounted a view doesn't seem right.

Reply:
Yeah, I agree that this can be more consistent. Do you think that there should be one loading state to rule the entire app (ie. `app.isLoading`) or should there be a loading state for every view (ie. `loginForm.isLoading`, `passwordForm.isLoading`, etc.)? I kind of think the one loading state would be OK for now, and it could eventually turn into the equivalent of `UIUtils.Loading.show()` and `UIUtils.Loading.hide()` from web-e which seems to work pretty well.

Reply:
I was thinking the loading state would be tied to the data model that you are waiting to load...

So e.g. when entering your `email` step we would: 
1. Set `account.loading` 
2. Call `GetAccountStatus` 
3. Set `account.loading` to false once the data is available

A simpler approach could be taken with error states.

Reply:
In that case, then don't you think it's a waste of logic to set the loading state to `false,` knowing that the view is immediately going to be unmounted? In other words, there would be no UI to show that `account.loading` is false again, so it's just wasted data sitting in Ion with no views bound to it.

Reply:
I can see that perspective, but I think it's not a waste since we maybe can't or should not assume that the view will immediately be unmounted and that nothing will ever go wrong with this step? 

We also are creating another way to do something when perhaps it would be better to standardize on one single way. 

Also small data like `loading: false` is very cheap. If that was a concern we could go from `loading: true` to no loading flag at all. But I don't think that's a problem we'd ever have.

Reply:
OK, thanks. As I was thinking about how this multi-step view could be refactored at some point, I was actually thinking that it would be good to tie the loading setting to the entire set of steps. So if we had several multi-step views, each of those flows would have its own loading setting in Onyx. Would you be OK with that? I think it's a best of both worlds where we don't have everything using a global setting, but also each individual form isn't using it's own personal setting either.

Reply:
Sure that sounds ok for now. 

I do think it would be beneficial to maybe start a wider convo about how handling loading and error states in general should work in the app. My main concern would be having a bunch of different ways to do it. It's an interesting topic, but one that we haven't had to deal with outside of the sign in flow (I suspect this will change).

Reply:
Cool, I created an issue to get to this. Thanks!

Reply:
(I meant to refactor `isLoading`... I didn't create an issue to have that bigger discussion)

Reply:
Sounds good! We got some time to hash that stuff out. 

FWIW, there was a decent conversation about it around the time when this was created --> https://github.com/Expensify/Expensify.cash/pull/556/files so my suggestion was based on that precedent.